tools/ai-evals/azsdk-mcp/Evaluators/ExpectedToolInputEvaluator.cs

```diff
@@ -0,0 +1,152 @@
+
+                return result;
+            }
+
+            // Get tool calls to compare them
+            var expectedToolCalls = GetToolContent(context.ChatMessages);
+            var actualToolCalls = GetToolContent(modelResponse.Messages);
+
+            if (!expectedToolCalls.Any())
+            {
+                metric.AddDiagnostics(
+                    EvaluationDiagnostic.Error(
+                        $"No Tool calls detected inside of {nameof(additionalContext)} collection."));
+
+                return result;
+            }
+
+            if (!actualToolCalls.Any())
+            {
+                metric.AddDiagnostics(
+                    EvaluationDiagnostic.Error(
+                        $"Provided LLM Result contained no tool calls."));
+
+                return result;
+            }
+            var expCount = expectedToolCalls.Count();
+            var actCount = actualToolCalls.Count();
+            if (expCount != actCount)
+            {
+                metric.AddDiagnostics(
+                    EvaluationDiagnostic.Error(
+                        $"The LLM Result had more tool calls than expected: Actual calls: {actCount}, Expected calls: {expCount}"));
+
+                return result;
+            }
+
+            var countCalls = 0;
+            foreach (var toolCall in expectedToolCalls.Zip(actualToolCalls, (exp, act) => new {exp, act}))
+            {
+                var expToolName = toolCall.exp.Name;
+                var actToolName = toolCall.act.Name;
+                countCalls++;
+
+                metric.AddDiagnostics(
+                    EvaluationDiagnostic.Informational(
+                        $"Tool call #{countCalls}, expected the {expToolName} tool, and LLM called the {actToolName} tool."));
+
+                // If the names do not align then the tool calls were made in the wrong order by the LLM
+                if (expToolName != actToolName)
+                {
+                    metric.AddDiagnostics(
+                    EvaluationDiagnostic.Error(
+                        $"Tool call made in the wrong order. Expected the {expToolName} tool but the LLM called {actToolName} tool. This was tool call #{countCalls}"));
+                    return result;
+                }
+
+                // Since it is a dictionary with a object may just be easier to serialize and compare strings. Can then even return the string args to the user if something went wrong. 
+                try
+                {
+                    // Serialize with consistent options for better comparison
+                    var options = new JsonSerializerOptions
+                    {
+                        WriteIndented = false,
+                        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+                    };
+
+                    var expectedJson = JsonSerializer.Serialize(toolCall.exp.Arguments, options);
```

Comment:
Probably a rare case these days, but if you can use a sorted dictionary it will avoid any potential flaky tests as dictionary key ordering is not guaranteed.